packages/blocks/src/api/raw-handling/special-comment-converter.js

```diff
@@ -69,11 +69,7 @@ function moreCommentConverter( node, doc ) {
 	// If our `<!--more-->` comment is in the middle of a paragraph, we should
 	// split the paragraph in two and insert the more block in between. If not,
 	// the more block will eventually end up being inserted after the paragraph.
-	if (
-		! node.parentNode ||
-		node.parentNode.nodeName !== 'P' ||
-		node.parentNode.childNodes.length === 1
-	) {
+	if ( ! node.parentNode || node.parentNode.nodeName !== 'P' ) {
```

Comment:
@ellatrix: can you explain this change? The code no longer matches the comment. If you can confirm my intuition here:

* Now the `else` branch will be run whenever the comment node's parent is a paragraph, even if it only has one child (the comment node).
* This forces all paragraphs through the `paragraphBuilder` walk, ensuring that paragraphs that only contained the comment (and would become `<p></p>`) are destroyed.

If the above is right, some follow-up questions:

* Is it necessary to the current PR, or an adjacent change?
* Looking at the tests in `packages/blocks/src/api/raw-handling/test/special-comment-converter.js`, there is now some inconsistency between the handling of `<!--more-->` and `<!--nextpage-->` — with the latter, tests still expect empty paragraph nodes to be produced. Should we fix that?

Reply:
Not adjacent, it was required to update this because some of the integration tests started failing with extra empty paragraphs that the original change was no longer removing.

I didn't think the comment no longer matched, but I reworded it.

I also fixed it for next page.